assets/js/checkout/checkout.js

```diff
@@ -59,7 +59,7 @@ angular.module(moduleName, ['credit-cards', 'angular.filter'])
                         $scope.checkout.billingAddressEqualsShipping = cart.hasPhysicalProducts && !angular.isObject($scope.checkout.payment.billingAddress);
 
                         $scope.checkout.canCartBeRecurring = $scope.customer.isRegisteredUser && _.all(cart.items, function(x) { return !x.isReccuring });
-                        $scope.checkout.paymentPlan = cart.paymentPlan && _.findWhere($scope.checkout.availablePaymentPlans, { intervalCount: cart.paymentPlan.intervalCount, interval: cart.paymentPlan.interval }) ||
+                        $scope.checkout.paymentPlan = cart.paymentPlan && _.findWhere($scope.checkout.availablePaymentPlans, { intervalCount: cart.paymentPlan.intervalCount, interval: cart.paymentPlan.interval.toLowerCase() }) ||
```

Comment:
To avoid such errors in maybe it makes sense to make sure $scope.checkout.availablePaymentPlans items 'interval' property is in the lower case too.